### ISSUE TITLE
How did you evaluate the NME of your model and PRNet in 3D face reconstruction?

### ISSUE THREAD

**Issue by deepdarkfans** [XgTu/2DASL#33]
PRNet has different dimensions than any other model because the neck part is removed. When I evaluated PRNet and your model, I found that PRNet's has the worst NME, probably because I removed some dimensions from the ground truth, so how did you evaluate the two models?

**Comment by Nancyh-e** [comment]
hi, how to get NME of PRNet，i get a bigger result compare to the NME paper. hope your reply!thanks!

**Comment by HOMGH** [comment]
Hi @deepdarkfans , did you get your answer? I have the same question as we get 43K vertices using PRNet but the groundtruth for AFLW2000 data set has 53K vertices.

**Comment by deepdarkfans** [comment]
> Hi @deepdarkfans , did you get your answer? I have the same question as we get 43K vertices using PRNet but the groundtruth for AFLW2000 data set has 53K vertices.

No, but I have a method you can try. When calculating NME, you can multiply the ground truth of AFLW2000 by the weight matrix of the PRNet loss function, which should ignore the neck part

**Comment by deepdarkfans** [comment]
> hi, how to get NME of PRNet，i get a bigger result compare to the NME paper. hope your reply!thanks!

 I have a method you can try. When calculating NME, you can multiply the ground truth of AFLW2000 by the weight matrix of the PRNet loss function, which should ignore the neck part

**Comment by HOMGH** [comment]
> > Hi @deepdarkfans , did you get your answer? I have the same question as we get 43K vertices using PRNet but the groundtruth for AFLW2000 data set has 53K vertices.
> 
> No, but I have a method you can try. When calculating NME, you can multiply the ground truth of AFLW2000 by the weight matrix of the PRNet loss function, which should ignore the neck part

Thank you for your reply. Where can I find this weight matrix? Do you mean the uv_face _mask in ./Data folder? (below)
But this can be multiplied by the uv position map not the vertices. So should we first convert AFLW2000 ground truth to uv position map and then multiply with this map?
![image](https://user-images.githubusercontent.com/65192528/126371844-e7592f2c-8e3f-4c33-b7cf-8c8b106fb8e0.png)

**Comment by deepdarkfans** [comment]
Yes, you need to convert AFLW2000 to UV position map,

**Comment by HOMGH** [comment]
> Yes, you need to convert AFLW2000 to UV position map,

Awesome! Thanks :)
